internal/k8stest/k8s_data_helpers.go

```diff
@@ -0,0 +1,194 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package k8stest // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8stest"
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	Equal = iota
+	Regex
+	Exist
+)
+
+type ExpectedValue struct {
+	mode  int
+	value string
+}
+
+func NewExpectedValue(mode int, value string) *ExpectedValue {
+	return &ExpectedValue{
+		mode:  mode,
+		value: value,
+	}
+}
```

Comment:
This code has nothing to do with k8s. Let's keep it in the e2e test. Ideally will need to be replaced with the pdatatest package going forward

Reply:
The problem is that the `ScanTracesForAttributes`, `ScanMetricsForAttributes`, and `ScanLogsForAttributes` functions all use these values. Should those just be left in its original location since they're also not k8s specific?

Reply:
Yes, all of them should stay there as well, they are not related to k8s

Reply:
I've moved all functions unrelated to k8s back to their original location, as far as I can tell.